glib-sys/src/lib.rs

```diff
@@ -1589,7 +1589,8 @@ bitflags! {
 // Unions
 pub type GDoubleIEEE754 = c_void; // union
 pub type GFloatIEEE754 = c_void; // union
-pub type GMutex = c_void; // union
+#[repr(C)]
+pub struct GMutex(u64); // union
```

Comment:
I just noticed that this is not 100% correct with the alignment on 32 bit architectures. New pull request in ~1 hour when I'm back home